src/libraries/illink-sharedframework.targets

```diff
@@ -16,26 +16,15 @@
       <!-- suppress warnings with the following codes:
            IL2009: Could not find method A in type B specified in resource C
            IL2025: Duplicate preserve of A in B
-           IL2026: Calling A which has B can break functionality when trimming application code. The target method might be removed.
            IL2035: Unresolved assembly A in DynamicDependencyAttribute on B
-           IL2050: P/invoke method A declares a parameter with COM marshalling. Correctness of COM interop
-                   cannot be guaranteed after trimming. Interfaces and interface members might be removed.
       -->
-      <LinkerNoWarn>IL2009;IL2025;IL2026;IL2035;IL2050</LinkerNoWarn>
-      <!-- IL2032,IL2055,IL2057-IL2061: Reflection intrinsics with unknown arguments -->
-      <LinkerNoWarn>$(LinkerNoWarn);IL2032;IL2055;IL2057;IL2058;IL2059;IL2060;IL2061</LinkerNoWarn>
-      <!-- IL2062-IL2066: Unknown values passed to locations with DynamicallyAccessedMemberTypes -->
-      <LinkerNoWarn>$(LinkerNoWarn);IL2062;IL2063;IL2064;IL2065;IL2066</LinkerNoWarn>
-      <!-- IL2067-IL2091: Unsatisfied DynamicallyAccessedMembers requirements -->
-      <LinkerNoWarn>$(LinkerNoWarn);IL2067;IL2068;IL2069;IL2070;IL2071;IL2072;IL2073;IL2074;IL2075;IL2076;IL2077;IL2078;IL2079;IL2080;IL2081;IL2082;IL2083;IL2084;IL2085;IL2086;IL2087;IL2088;IL2089;IL2090;IL2091</LinkerNoWarn>
+      <LinkerNoWarn>IL2009;IL2025;IL2035</LinkerNoWarn>
```

Comment:
I guess I would have expected there to be no warnings that are suppressed globally. Why do we still have these?

Reply:
The remaining global warnings cannot be suppressed with suppression attributes as they have to do with incorrect data/annotations/descriptors in the XML files we pass to the linker, rather than warnings about reflection usages. They need to be fixed rather than suppressed. I'll add a task to address this in a follow up to https://github.com/dotnet/runtime/issues/38033.


On IL2025 in https://github.com/dotnet/runtime/pull/40691#discussion_r495499750:

>They are all coming from analysis on the embedded `System.Private.CoreLib.xml` file in the input `System.Private.CoreLib` assembly - https://gist.github.com/layomia/2f2d007b9bac71c32fa0aff0e5832ec8. I'm not entirely sure - does this mean that we can remove some of the entries in the CoreLib descriptor XML files due to linker annotations present in the various libraries? Will reach out offline to discuss.